src/geometry.rs

```diff
@@ -800,4 +822,46 @@ mod tests {
             normals.clone(),
         );
     }
+
+    #[test]
+    fn test_orphan_vertex_detection() {
```

Comment:
For future reference, it is better to cover one test case with one test function for better granularity. If this test fails, one does not know exactly why immediately and has to do more investigation to find out. Same for: `test_orphan_normal_detection`